tcpkeepalive_darwin.go

```diff
@@ -0,0 +1,34 @@
+package tcpkeepalive
+
+import (
+	"os"
+	"runtime"
+	"syscall"
+	"time"
+)
+
+const sysTCP_KEEPINTVL = 0x101
+
+func setIdle(fd uintptr, d time.Duration) error {
+	// not possible with darwin
+	return nil
+}
+
+func setCount(fd uintptr, n int) error {
+	// not possible with darwin
+	return nil
+}
+
+func setInterval(fd uintptr, d time.Duration) error {
+	// # from https://golang.org/src/net/tcpsockopt_darwin.go
+	d += (time.Second - time.Nanosecond)
+	secs := int(d.Seconds())
+	switch err := syscall.SetsockoptInt(int(fd), syscall.IPPROTO_TCP, sysTCP_KEEPINTVL, secs); err {
+	case nil, syscall.ENOPROTOOPT: // OS X 10.7 and earlier don't support this option
+	default:
+		return os.NewSyscallError("setsockopt", err)
+	}
+	err := syscall.SetsockoptInt(int(fd), syscall.IPPROTO_TCP, syscall.TCP_KEEPALIVE, secs)
```

Comment:
From the man https://github.com/apple/darwin-xnu/blob/5394bb038891708cd4ba748da79b90a33b19f82e/bsd/man/man4/tcp.4#L172 TCP_KEEPALIVE should define the interval so it doesn't belong here but should be put in the `setIdle` function.